extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java

```diff
@@ -141,13 +142,15 @@ public OidcClient apply(String tokenRequestUri, Throwable t) {
                                 : OidcConstants.PASSWORD_GRANT;
                         setGrantClientParams(oidcConfig, tokenGrantParams, grantType);
 
+                        Map<String, String> grantOptions = oidcConfig.getGrantOptions()
+                                .get(oidcConfig.grant.getType().name().toLowerCase());
                         if (oidcConfig.grant.getType() == Grant.Type.PASSWORD) {
-                            Map<String, String> passwordGrantOptions = oidcConfig.getGrantOptions()
-                                    .get(OidcConstants.PASSWORD_GRANT);
                             tokenGrantParams.add(OidcConstants.PASSWORD_GRANT_USERNAME,
-                                    passwordGrantOptions.get(OidcConstants.PASSWORD_GRANT_USERNAME));
+                                    grantOptions.get(OidcConstants.PASSWORD_GRANT_USERNAME));
                             tokenGrantParams.add(OidcConstants.PASSWORD_GRANT_PASSWORD,
-                                    passwordGrantOptions.get(OidcConstants.PASSWORD_GRANT_PASSWORD));
+                                    grantOptions.get(OidcConstants.PASSWORD_GRANT_PASSWORD));
+                        } else if (grantOptions != null && oidcConfig.grant.getType() == Grant.Type.CLIENT) {
```

Comment:
Just a note to myself - throw the configuration exception if the password options (in the branch above) if the options are null.